x/attribute/keeper/keeper.go

```diff
@@ -78,20 +79,24 @@ func (k Keeper) GetAttributes(ctx sdk.Context, acc sdk.AccAddress, name string)
 }
 
 // IterateRecords iterates over all the stored attribute records and passes them to a callback function.
-func (k Keeper) IterateRecords(ctx sdk.Context, handle Handler) error {
+func (k Keeper) IterateRecords(ctx sdk.Context, prefix []byte, handle Handler) error {
```

Comment:
Having this prefix param here kind of leaks some implementation details out of the keeper that I am not sure we really should...

Reply:
we can remove this after we do the conversion, but technically an iterate handler could be using to iterate over different prefixes(rather keys, like the name module) without writing methods for each, i thought having a generic handler already violates the encapsulation principle here, meaning that the caller is responsible for how/what they iterate/filter through..but for this one we can remove the param after this conversion..or we can write the a custom method for that prefix(dup code, but again can be removed)..